python/erdos/__init__.py

```diff
@@ -61,85 +95,98 @@ def connect(op_type, config, read_streams, *args, **kwargs):
         elif isinstance(stream, ReadStream):
             py_read_streams.append(stream._py_read_stream)
         else:
-            raise TypeError("Unable to convert {stream} to ReadStream".format(
-                stream=stream))
+            raise TypeError(
+                "Unable to convert {stream} of type {stream_type} to ReadStream"
+                .format(stream=stream, stream_type=type(stream)))
 
     internal_streams = _internal.connect(op_type, config, py_read_streams,
                                          args, kwargs, node_id)
+    logger.debug("Converting {} write stream(s) returned by "
+                 "operator {} to read stream(s).".format(
+                     len(internal_streams), config.name))
     return [ReadStream(_py_read_stream=s) for s in internal_streams]
 
 
 def reset():
-    """Resets internal seed and creates a new dataflow graph.
+    """Create a new dataflow graph.
 
-    Note that no streams or operators can be re-used safely.
+    Note:
+        A call to this function renders the previous dataflow graph unsafe to 
+        use.
     """
+    logger.info("Resetting the default graph.")
     global _num_py_operators
     _num_py_operators = 0
     _internal.reset()
 
 
-def run(graph_filename=None, start_port=9000):
+# TODO (Sukrit) : Should this be called a GraphHandle?
```

Comment:
I think we might have to do a bit of work to correctly shutdown the dataflow. AFAIk, the NodeHandle is only terminating the processes running on the current machine.

@pschafhalter Do you agree? Should we open an issue about this so that we don't forget about this issue.

Reply:
Yes, let's make an issue. Although currently we don't expose a way to distribute pipelines across multiple machines when using Python, so in a way this isn't a problem right now.